mdata/aggmetric.go

```diff
@@ -332,18 +252,12 @@ func (a *AggMetric) addAggregators(ts uint32, val float64) {
 
 // write a chunk to persistent storage. This should only be called while holding a.Lock()
 func (a *AggMetric) persist(pos int) {
-	if !cluster.Manager.IsPrimary() {
-		if LogLevel < 2 {
-			log.Debug("AM persist(): node is not primary, not saving chunk.")
-		}
-		return
-	}
-
-	pre := time.Now()
 	chunk := a.Chunks[pos]
+	pre := time.Now()
 
-	if chunk.Saved || chunk.Saving {
-		// this can happen if chunk was persisted by GC (stale) and then new data triggered another persist call
```

Comment:
Now that chunk saving has been decoupled from chunks being closed (chunk.Finish()) the old comment is incorrect and the new one is correct.

persist() is only called when a node is a primary, the chunk is not already closed and
a) a new metric has been received for the next chunk
b) GC has determined the chunk is stale and it is not already 

Before persist is called, the chunk is closed, which will prevent any future presist() calls from being made.

So, the only way persist can be called on a chunk that is already saved is if there are 2 primary MT nodes both saving chunks to Cassandra or a primary failed and this node was promoted to be primary but metric consuming is lagging. ie exactly what the new comment states.